### ISSUE TITLE
Question: useRevalidate hook

### ISSUE THREAD

**Issue by JClackett** [sergiodxa/remix-utils#32]
Quick backstory.
I am building a radio web app and have a page where I can like the episode, but also have a global audio player that sits on top of every route (like Soundcloud). I have a like button on both the episode detail page and on the player. The player, being a global component which can be rendered on any page, obviously has to fetch its data using useFetcher. 

When I like the episode on the player, I would like to update the detail page's "like" button to reflect the change too, so my idea was on response from the useFetcher I use the revalidate hook to refetch the current pages loaders. However when using the useRevalidate hook it just navigates back to the home page i.e. "localhost:3000" every time, is there something I'm missing? Is there a better way to do what I need? I managed to get the functionality I wanted by directly using useNavigate + current location like below, but am getting some weird stuff where im now getting double flash messages, so feel like im doing something a bit weird still.

```ts
const location = useLocation()
const navigate = useNavigate()
React.useEffect(() => {
    if (savedSubmit.data?.success) {
      navigate(location.pathname, { replace: true })
    }
}, [savedSubmit.data, location.pathname])
```

**Comment by JClackett** [comment]
Actually, just completely removed the need for this!

I was originally posting to a resource route using useFetcher, but I moved the logic of saving an episode to the episode details action function, and made both "like" buttons actual forms that posted to this action. I then just checked useTransition in the global player component and the detail page to see if we are posting to this action and updated some state for optimistic ui. Using a proper form meant that all the loaders "revalidated" themselves, no need to manually validate! Much cleaner :D 

I guess my question about your hook is still valid, which is, why does it navigate back to the homepage?
